### ISSUE TITLE
[BUG] gdf.sample(frac=1) shuffles data very poorly compared to df.sample(frac=1)

### ISSUE THREAD

**Issue by esnvidia** [rapidsai/cudf#8451]
**Describe the bug**
Dataframe sampling (as a proxy for shuffling a dataset since dataframe.shuffle does not exist in cudf yet) is working poorly compared to the pandas counterpart. It appears that the data might be moved around in chunks.

**Steps/Code to reproduce bug**
```
# make an imbalanced dataset
z = cupy.zeros(9900)
o = cupy.ones(100)
data = cupy.concatenate((z,o))

gdf = cudf.DataFrame(data, columns=['isFraud'])

# plot original data for reference
fig, ax = plt.subplots()
ax.scatter(range(len(gdf)), gdf['isFraud'].to_pandas())

# plot cudf sampled data
fig, ax = plt.subplots()
ax.scatter(range(len(gdf)), gdf.sample(frac=1, random_state=42)['isFraud'].to_pandas())

# plot pandas sampled data
pdf = gdf.to_pandas()
fig, ax = plt.subplots()
ax.scatter(range(len(pdf)), pdf.sample(frac=1, random_state=42)['isFraud'])

```

![shuffling_issue](https://user-images.githubusercontent.com/80840697/121082101-98b0f100-c7ab-11eb-8b07-cfd9f222d7df.png)

**Expected behavior**
Positive class to be more or less uniformly distributed after the sampling.

**Environment details (please complete the following information):**
conda rapids-0.19

**Comment by beckernick** [comment]
I'm not able to reproduce this issue in the current nightly. Would you be able to verify this still presents in an up to date environment?

```python
import cudf
import numpy as np
import cupy
import matplotlib.pyplot as plt

z = cupy.zeros(9900)
o = cupy.ones(100)
data = cupy.concatenate((z,o))

gdf = cudf.DataFrame(data, columns=['isFraud'])

# # plot original data for reference
# fig, ax = plt.subplots()
# ax.scatter(range(len(gdf)), gdf['isFraud'].to_pandas())

# plot cudf sampled data
fig, ax = plt.subplots()
ax.scatter(range(len(gdf)), gdf.sample(frac=1, random_state=42)['isFraud'].to_pandas())
ax.set_title("cudf sampled")

# plot pandas sampled data
pdf = gdf.to_pandas()
fig, ax = plt.subplots()
ax.scatter(range(len(pdf)), pdf.sample(frac=1, random_state=42)['isFraud'])
ax.set_title("pandas sampled")
```
![Screen Shot 2021-06-14 at 8 57 55 AM](https://user-images.githubusercontent.com/8457388/122485111-278bed80-cfa4-11eb-9759-c1468a81c0c0.png)

**Comment by esnvidia** [comment]
@beckernick Looks fixed now in 21.06. Not sure what fix was made.

**Comment by beckernick** [comment]
Great, glad things are now working. Closing this issue, but please feel free to reopen if this presents in the future.
